ipn/ipnlocal/local.go

```diff
@@ -1742,6 +1742,21 @@ func (b *LocalBackend) readPoller() {
 func (b *LocalBackend) WatchNotifications(ctx context.Context, mask ipn.NotifyWatchOpt, fn func(roNotify *ipn.Notify) (keepGoing bool)) {
 	ch := make(chan *ipn.Notify, 128)
 
+	redact := func(n *ipn.Notify) bool {
```

Comment:
This works, but the "if not redact" below reads weirdly. "You don't want me to redact? Did the redaction fail?"

Maybe instead here do something like:

```go
    origFn := fn
    if mask&ipn.NotifyNoPrivateKeys != 0 {
          fn = func(...) {
                 ... use origFn conditionally ...
          }
     }
```

... and then keep the callers below just calling `fn` ?

Reply:
Done.